### ISSUE TITLE
Enable implicit RID with `--sc`

### ISSUE THREAD

**Issue by richlander** [dotnet/sdk#21405]
The new `--sc` syntax works for self-contained apps, per https://github.com/dotnet/sdk/pull/19681.

It still requires a RID. Let's enable it to work with the implicit SDK RID so that we don't create a breaking change in .NET 7. This argument is new. It also will enable users to get around the new warning with `-r`.

This change is some really nice fit-and-finish for .NET 6. This behavior is the original intent of https://github.com/dotnet/sdk/issues/19576.

**Comment by marcpopMSFT** [comment]
@richlander I moved this to 6.0.2xx since we're closing down on 1xx. I don't see it as a breaking change to enable this in 2xx since it's loosening the limitations of the CLI. Let me know if you disagree.

**Comment by richlander** [comment]
It'd be really nice to get this in .100 to make it easier for folks to deal with NETSDK1179. They go together very nicely.

We're also not done with NET1179 https://github.com/dotnet/sdk/issues/21560

That said, if we cannot make it, .200 is fine. Yes, you are right. It isn't breaking. I was wrong about that.

**Comment by richlander** [comment]
Fair enough. Let's make the change in .NET 7 as you say. I changed the milestone.
